relay/polkadot/tests/ahm/mod.rs

```diff
@@ -48,3 +49,101 @@ async fn remote_ext_test_setup() -> Option<RemoteExternalities<Block>> {
 
 	Some(ext)
 }
+
+#[test]
+fn call_filter_works() {
+	let mut t: sp_io::TestExternalities =
+		frame_system::GenesisConfig::<T>::default().build_storage().unwrap().into();
+
+	// MQ calls are never filtered:
+	let mq_call =
+		polkadot_runtime::RuntimeCall::MessageQueue(pallet_message_queue::Call::<T>::reap_page {
+			message_origin: AggregateMessageOrigin::Ump(
+				runtime_parachains::inclusion::UmpQueueId::Para(ParaId::from(1000)),
+			),
+			page_index: 0,
+		});
+	// System calls are filtered during the migration:
+	let system_call =
+		polkadot_runtime::RuntimeCall::System(frame_system::Call::<T>::remark { remark: vec![] });
+	// Indices calls are filtered during and after the migration:
+	let indices_call =
+		polkadot_runtime::RuntimeCall::Indices(pallet_indices::Call::<T>::claim { index: 0 });
+
+	let is_allowed = |call: &polkadot_runtime::RuntimeCall| Pallet::<T>::contains(call);
+
+	// Try the BaseCallFilter
+	t.execute_with(|| {
+		// Before the migration starts
+		{
+			RcMigrationStage::<T>::put(MigrationStage::Pending);
+
+			assert!(is_allowed(&mq_call));
+			assert!(is_allowed(&system_call));
+			assert!(is_allowed(&indices_call));
+		}
+
+		// During the migration
+		{
+			RcMigrationStage::<T>::put(MigrationStage::ProxyMigrationInit);
+
+			assert!(is_allowed(&mq_call));
+			assert!(!is_allowed(&system_call));
+			assert!(!is_allowed(&indices_call));
+		}
+
+		// After the migration
+		{
+			RcMigrationStage::<T>::put(MigrationStage::MigrationDone);
+
+			assert!(is_allowed(&mq_call));
+			assert!(is_allowed(&system_call));
+			assert!(!is_allowed(&indices_call));
+		}
+	});
+
+	// Try to actually dispatch the calls
+	t.execute_with(|| {
+		let alice = AccountId32::from([0; 32]);
+		<pallet_balances::Pallet<T> as frame_support::traits::Currency<_>>::deposit_creating(
+			&alice,
+			u64::MAX.into(),
+		);
+
+		// Before the migration starts
+		{
+			RcMigrationStage::<T>::put(MigrationStage::Pending);
+
+			assert!(!is_forbidden(&mq_call));
+			assert!(!is_forbidden(&system_call));
+			assert!(!is_forbidden(&indices_call));
+		}
+
+		// During the migration
+		{
+			RcMigrationStage::<T>::put(MigrationStage::ProxyMigrationInit);
+
+			assert!(!is_forbidden(&mq_call));
+			assert!(is_forbidden(&system_call));
+			assert!(is_forbidden(&indices_call));
+		}
+
+		// After the migration
+		{
+			RcMigrationStage::<T>::put(MigrationStage::MigrationDone);
+
+			assert!(!is_forbidden(&mq_call));
+			assert!(!is_forbidden(&system_call));
+			assert!(is_forbidden(&indices_call));
+		}
+	});
+}
+
+fn is_forbidden(call: &polkadot_runtime::RuntimeCall) -> bool {
+	let Err(err) = call.clone().dispatch(RuntimeOrigin::signed(AccountId32::from([0; 32]))) else {
```

Comment:
This is actually a nice test to run on basically every pallet and extrinsic during the migration except for a few exceptions that are explicitly allowed from signed origins - the ones that are root only will fail anyway from this origin.

We could then have a separate check for the root calls that should be filtered, with the baseline being that all root calls are allowed except the ones we know will cause problems.

Reply:
Yes we could do that. Currently we need to manually construct the call arguments as well since our RuntimeCall enum does not implement an `Arbitrary` trait.